### PR TITLE
[qat] LinearReLU module

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack:
&nbsp;&nbsp;&nbsp;&nbsp;:black_circle:&nbsp; **#23022 [qat] LinearReLU module**&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D16286800/)
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #23008 [qat] ConvReLU2d module&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D16286670/)

will be tested in later diffs.

Added LinearReLU module for qat, allows conversion from torch.nn._intrisic.LinearReLU to torch.nn._intrinsic.qat.LinearReLU

Differential Revision: [D16286800](https://our.internmc.facebook.com/intern/diff/D16286800/)

### CODE REVIEW

File: torch/nn/_intrinsic/qat/modules/linear_relu.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+from torch.nn._intrisic import LinearReLU as NNLinearReLU
+from torch.nn.qat import Linear as QATLinear
+from torch.quantization.QConfig import default_qat_qconfig
+import torch.nn.functional as F
+
+class LinearReLU(QATLinear):
+    r"""
+    A linear module attached with FakeQuantize modules for both output
```
Comment (jianyuh) [inline torch/nn/_intrinsic/qat/modules/linear_relu.py]: A LinearRelu module (linear fused with relu)

File: torch/nn/_intrinsic/qat/modules/linear_relu.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    Examples::
+
+        >>> m = nn.qat.LinearReLU(20, 30)
+        >>> input = torch.randn(128, 20)
+        >>> output = m(input)
+        >>> print(output.size())
+        torch.Size([128, 30])
+    """
+    __constants__ = ['bias', 'in_features', 'out_features']
```
Comment (dzhulgakov) [inline torch/nn/_intrinsic/qat/modules/linear_relu.py]: this is needed only for TorchScript, right? probably delete as it's not a weak module

**Review by jianyuh — APPROVED** [review]
LGTM! Please land after it passes all OSS test.

File: torch/nn/_intrinsic/qat/modules/conv_relu.py [hunk 3]
```diff
@@ -10,9 +10,7 @@ class ConvReLU2d(QATConv2d):
     FakeQuantize modules for both output activation and weight for
     quantization aware training.
 
-    We adopt the same interface as `torch.nn.Conv2d`, please see
-    https://pytorch.org/docs/stable/nn.html?highlight=conv2d#torch.nn.Conv2d
-    for documentation.
+    We adopt the same interface as :class:`~torch.nn.Conv2d`.
```
Comment (raghuramank100) [inline torch/nn/_intrinsic/qat/modules/conv_relu.py]: NIT: Fix comment

File: torch/nn/_intrinsic/qat/modules/linear_relu.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        >>> output = m(input)
+        >>> print(output.size())
+        torch.Size([128, 30])
+    """
+    __FLOAT_MODULE__ = NNLinearReLU2d
+
+    def __init__(self, in_features, out_features, bias=True,
+                 activation_fake_quant=default_qat_qconfig.activation,
+                 weight_fake_quant=default_qat_qconfig.weight):
+        assert bias, 'nobias is not supported in qat LinearReLU module yet'
```
Comment (raghuramank100) [inline torch/nn/_intrinsic/qat/modules/linear_relu.py]: Why is this not supported?

Comment (jerryzh168) [reply torch/nn/_intrinsic/qat/modules/linear_relu.py]: ..don't remember why we have this, will remove

File: torch/nn/_intrinsic/qat/modules/conv_relu.py (hunk 3 above)
Comment (jerryzh168) [reply torch/nn/_intrinsic/qat/modules/conv_relu.py]: this is conv module

Comment (raghuramank100) [reply torch/nn/_intrinsic/qat/modules/conv_relu.py]: Sorry, my bad

**Review by raghuramank100 — APPROVED** [review]

**Pull request closed without merging**